video/out/vo_sixel.c

```diff
@@ -0,0 +1,483 @@
+                                     LARGE_NORM, REP_CENTER_BOX,
+                                     QUALITY_LOW);
+    if (SIXEL_FAILED(status))
+        return status;
+
+    if (detect_scene_change(vo)) {
+        if (priv->dither)
+            sixel_dither_unref(priv->dither);
+
+        priv->dither = priv->testdither;
+        status = sixel_dither_new(&priv->testdither, priv->opts->reqcolors, NULL);
+
+        if (SIXEL_FAILED(status))
+            return status;
+        sixel_dither_set_diffusion_type(priv->dither, priv->opts->diffuse);
+    } else {
+        sixel_dither_set_body_only(priv->dither, 1);
+    }
+
+    return status;
+}
+
+static int resize(struct vo *vo)
+{
+    struct priv *priv = vo->priv;
+
+    talloc_free(priv->buffer);
+
+    if (priv->dither) {
+        sixel_dither_unref(priv->dither);
+        priv->dither = NULL;
+    }
+
+    if (priv->testdither) {
+        sixel_dither_unref(priv->testdither);
+        priv->testdither = NULL;
+    }
+    SIXELSTATUS status = sixel_dither_new(&priv->testdither, priv->opts->reqcolors, NULL);
+
+    if (SIXEL_FAILED(status)) {
+        return status;
+    }
+    priv->buffer =
+        talloc_array(NULL, uint8_t, depth * priv->opts->width * priv->opts->height);
+
+    return 0;
+}
+
+static int reconfig(struct vo *vo, struct mp_image_params *params)
+{
+    struct priv *priv = vo->priv;
+    priv->image_height = params->h;
+    priv->image_width  = params->w;
+    priv->image_format = params->imgfmt;
+    scroll_on_demand(vo);
+
+    priv->sws->src = *params;
+    priv->sws->dst = (struct mp_image_params) {
+        .imgfmt = IMGFMT,
+        .w = priv->opts->width,
+        .h = priv->opts->height,
+        .p_w = 1,
+        .p_h = 1,
+    };
+
+    priv->frame = mp_image_alloc(IMGFMT, priv->opts->width, priv->opts->height);
+    if (!priv->frame)
+        return -1;
+
+    if (mp_sws_reinit(priv->sws) < 0)
+        return -1;
+
+    printf(ESC_HIDE_CURSOR);
+    printf(ESC_CLEAR_SCREEN);
+    vo->want_redraw = true;
+
+    return resize(vo);
+}
+
+static void draw_image(struct vo *vo, mp_image_t *mpi)
+{
+    struct priv *priv = vo->priv;
+    struct mp_image src = *mpi;
+    SIXELSTATUS status = SIXEL_FALSE;
+
+    mp_sws_scale(priv->sws, priv->frame, &src);
+
+    memcpy_pic(priv->buffer, priv->frame->planes[0], priv->opts->width * depth, priv->opts->height,
+               priv->opts->width * depth, priv->frame->stride[0]);
+
+    if (priv->opts->fixedpal) {
+        status = prepare_static_palette(vo);
+    } else {
+        status = prepare_dynamic_palette(vo);
+    }
+    if (SIXEL_FAILED(status)) {
+        return;
+    }
+
+    talloc_free(mpi);
+}
+
+static int sixel_write(char *data, int size, void *priv)
+{
+    return fwrite(data, 1, size, (FILE *)priv);
+}
+
+static void flip_page(struct vo *vo)
+{
+    struct priv* priv = vo->priv;
+    FILE* sixel_output_file = stdout;
+    SIXELSTATUS status = SIXEL_FALSE;
```

Comment:
```
../video/out/vo_sixel.c: In function ‘flip_page’:
../video/out/vo_sixel.c:385:17: warning: variable ‘status’ set but not used [-Wunused-but-set-variable]
  385 |     SIXELSTATUS status = SIXEL_FALSE;
      |                 ^~~~~~
```

with:  `gcc (GCC) 10.2.0`

Reply:
Have removed that variable and ignored the return value, so the warning doesn't show up now